### PR TITLE
Redesign home screen implementation

### PR DESCRIPTION
**Pull request by osamaasifoak**
Hey 👋 this PR contains

-  home screen API implementation
-  Web view scrolling issue fixed

### CODE REVIEW

File: lib/providers/folder/folder_provider.dart [hunk 1]
```diff
@@ -9,6 +9,7 @@ Future<FolderModel> folders(
   required int folderId,
 }) {
   final folderRepository = ref.watch(folderRepositoryProvider);
+  ref.keepAlive();
```
Comment (michaelcspeed) [inline lib/providers/folder/folder_provider.dart:12]: Why are these needed? Genuine question, I have no idea

**Review by michaelcspeed — APPROVED** [review]

File: lib/providers/folder/folder_provider.dart (hunk 1 above)
Comment (osamaasifoak) [reply lib/providers/folder/folder_provider.dart:12]: Hey @michaelcspeed, good question, please review the link for a better explanation. https://riverpod.dev/docs/concepts/modifiers/auto_dispose#refkeepalive

Comment (michaelcspeed) [reply lib/providers/folder/folder_provider.dart:12]: Cool thanks!
